### ISSUE TITLE
[Feature] Smart JSON substitution casting to null

### ISSUE THREAD

**Issue by riodeuno** [appsmithorg/appsmith#3811]
## Summary
`undefined` is considered incorrect in JSON, so we should have it automatically substituted to `null` during Smart JSON substitution

**Comment by trishaanand** [comment]
@riodeuno @Nikhil-Nandagopal : The real behaviour indeed is to set it to null. When the API executes, smart substitution ensures that null values are set correctly and the JSON sent out is valid.
Do you mean that the client should display this ? @hetunandu : Is this possible in smart substitution to handle undefined as null values?

**Comment by riodeuno** [comment]
@trishaanand Yes. We should try to maintain the accuracy of the displayed evaluated values. If we're executing the API with the values substituted as `null` we should show this in the evaluated values display.
